recipes.js

```diff
@@ -8,3 +8,97 @@ mongoose.connect('mongodb://localhost/recipeApp')
   }).catch(err => {
     console.error('Error connecting to mongo', err);
   });
+
+const recipeSchema = new Schema({
+	title: { 
+		type: String,
+		required: true,
+		unique: true,
+	},
+	level: {
+		type: String,
+		enum: [ "Easy Peasy", "Amateur Chef", "UltraPro Chef" ],
+	},
+	ingredients: [ String ],
+	cuisine: {
+		type: String,
+		required: true,
+	},
+	dishtype: {
+		type: String,
+		enum: [ "Breakfast", "Dish", "Snack", "Drink", "Dessert", "Other" ],
+	},
+	image: {
+		type: String,
+		default: "https://images.media-allrecipes.com/images/75131.jpg",
+	},
+	duration: {
+		type: Number,
+		min: 0,
+	},
+	creator: String,
+	created: {
+		type: Date,
+		default: Date.now
+	}
+});
+
+const Recipe = mongoose.model("Recipe", recipeSchema);
+
+
+
+Recipe.create({
+	title: "Food Porn",
+	level: "Easy Peasy",
+	ingredients: ["ing 1", "ing 2"],
+	cuisine: "American",
+	dishtype: "Breakfast",
+	image: "https://cdn1.opnminded.com/wp-content/uploads/2017/11/food-porn-03.jpg",
+	duration: 15,
+	creator: "JN",	
+})
+	.then(recipeDoc => {
+		console.log(recipeDoc.title);
+	})
+	.catch(err => {
+		console.log("recipe.create() FAILURE", err);
+	});
+
+Recipe.insertMany(data)
+	.then(list => {
+		list.forEach(recipe => {
+			console.log("recipe.insertMany() SUCCESS", list);
+		});
+	})
+	.catch(err => {
+		console.log("recipe.insertMany() FAILURE", err);
+	});
+
+Recipe.findOneAndUpdate(
+	{ title: { $eq: "Rigatoni alla Genovese" } },
+	{ $set: { duration: 100 } }
+	)
+	.then(recipe => {
+		console.log("recipe update SUCCESS");
+	})
+	.catch(err => {
+		console.log("recipe update FAILURE", err);
+	});
+
+
+Recipe.findOneAndRemove( { title: { $eq: "Carrot Cake" } } )
+	.then(recipe => {
+		console.log("recipe remove SUCCESS");
+	})
+	.catch(err => {
+		console.log("recipe remove FAILURE", err);
+	});
+
+
+mongoose.connection.close()
+	.then(() => {
+		console.log("Properly disconnected from DB");
+	})
+	.catch(err => {
+		console.log("Error disconnecting from DB", err);
+	});
```

Comment:
Hehe si seulement l'itération 6 était aussi facile ! 
C'est un peu compliqué donc on attend pas de vous que vous sachiez le faire. Le problème c'est de gérer l'asynchronie de toutes ces requêtes. Là, toutes les requêtes que tu as écrites plus haut sont correctes indépendemment les unes des autres, mais elles sont lancées toutes en même temps et ne prennent pas le même temps. Donc par exemple, il va essayer de supprimer le carrot cake avant même de l'avoir créé. Et en l'occurence, fermer la connection est l'action la plus rapide, donc dans ton script ici, il va commencer par faire ça et donc toutes les requêtes à la db vont échouer. 

Les promesses, ça sert justement à forcer certaines parties du code à s'executer dans un ordre particulier. On peut les chinet, c'est à dire dans le then d'une requête, appeler une autre fonction qui renvoit aussi une promeesse et mettre son then à la suite. Si tu veux essayer de comprendre comment ça marche, tu peux checker le code d'Amine : 
https://github.com/ironhack-labs/lab-mongoose-recipes/pull/253/files